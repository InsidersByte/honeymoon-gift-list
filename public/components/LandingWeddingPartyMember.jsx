/* @flow */

import React from 'react';
// FIXME:FLOW need to fix import .styl
import css from './LandingWeddingPartyMember.styl';

type PropsType = {
    weddingPartyMember: {
        name: string,
        title: string,
        imageUrl: string,
        description: string,
    },
};

export default function LandingWeddingPartyMember({ weddingPartyMember: { name, title, imageUrl, description } }: PropsType) {
    const backgroundImageStyle = { backgroundImage: `url(${imageUrl})` };

    return (
        <div className={css.root}>
            <div className={css.image} style={backgroundImageStyle} />

            <h1 className={css.name}>{name}</h1>

            <h2 className={css.title}>{title}</h2>

            <p className={css.description}>{description}</p>
        </div>
    );
}
