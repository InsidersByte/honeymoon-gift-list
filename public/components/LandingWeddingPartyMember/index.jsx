/* @flow */

import React from 'react';
import styled from 'styled-components';

type PropsType = {
  +name: string,
  +title: string,
  +imageUrl: string,
  +description: string,
};

const Root = styled.div`
  flex: 1 1 33%;
  margin-bottom: 3%;
  max-width: calc(100% / 3);
  padding: 6px 12px 0;

  @media (max-width: 1279px) {
    flex: 1 1 50%;
    max-width: calc(100% / 2);
  }

  @media (max-width: 736px) {
    flex: 1 1 100%;
    max-width: 100%;
  }
`;

const BackgroundImage = styled.div`
  background-image: url(${props => props.imageUrl});
  background-attachment: scroll;
  background-position: center center;
  background-repeat: no-repeat;
  background-size: cover;
  border-radius: 100%;
  height: 120px;
  margin: 0 auto;
  width: 120px;
`;

const Title = styled.h1`
  margin-top: 0;
`;

const SubTitle = styled.h3`
  margin-bottom: 0;
`;

const Description = styled.p`
  white-space: pre-wrap;
`;

const LandingWeddingPartyMember = ({ name, title, imageUrl, description }: PropsType) => (
  <Root>
    <BackgroundImage imageUrl={imageUrl} />
    <Title>{name}</Title>
    <SubTitle>"{title}"</SubTitle>
    <Description>{description}</Description>
  </Root>
);

export default LandingWeddingPartyMember;
