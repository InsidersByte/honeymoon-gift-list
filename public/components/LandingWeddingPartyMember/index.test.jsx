import React from 'react';
import { shallow } from 'enzyme';
import LandingWeddingPartyMember from './';

describe('LandingWeddingPartyMember', () => {
  it('should render correctly', () => {
    const weddingPartyMember = {
      name: 'Person',
      title: 'Title',
      imageUrl: 'http://image.png',
      description: 'this is a person',
    };

    const wrapper = shallow(<LandingWeddingPartyMember weddingPartyMember={weddingPartyMember} />);

    expect(wrapper).toMatchSnapshot();
  });
});
