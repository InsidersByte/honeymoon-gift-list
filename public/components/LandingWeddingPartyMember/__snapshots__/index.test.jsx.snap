// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingWeddingPartyMember should render correctly 1`] = `
<div
  className="root"
>
  <div
    className="image"
    style={
      Object {
        "backgroundImage": "url(http://image.png)",
      }
    }
  />
  <h1
    className="name"
  >
    Person
  </h1>
  <h3
    className="title"
  >
    "
    Title
    "
  </h3>
  <p
    className="description"
  >
    this is a person
  </p>
</div>
`;
